### PR TITLE
Python: remove `Verify` and `./diagnostics`

### PR DESCRIPTION
**Pull request by jjhenkel**
### Motivation and Context

In an effort to make the Python port more idiomatic, let's remove the `Verify` class and the `./diagnostics` sub-module. There are many more follow-on tasks to do here, but this is a good start (and already a large enough change).

### Description

This PR does the following:

1. Removes the `Verify` class, and re-writes all instances of `Verify.*` 
2. Adds a `validation.py` in the `./utils` sub-module to hand some of the more complex cases from `Verify` (checking that skills/functions/and function params have valid names)
3. Removes the rest of the `./diagnostics` sub-module (w/ a longer-term goal of removing all/most of our custom exception classes and, instead, using appropriate built-in error classes)

### Contribution Checklist

For Python, I've run `make pre-commit` to verify style/linting; I've run `pytest tests/`. No tests added here, this is a cosmetic/style change.

image

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anything :smile:

### CODE REVIEW

**Comment by jjhenkel** [comment]
One more small thing, just had to merge and `test_config.py` was out of compliance w/ `make pre-commit`. Brought it back into compliance, we should re-factor and standardize how we test skills (defined as files). (But that can come in a separate PR.)

File: python/semantic_kernel/kernel_extensions/inline_function_definitions.py [hunk 1]
```diff
@@ -12,6 +11,7 @@
 from semantic_kernel.semantic_functions.semantic_function_config import (
     SemanticFunctionConfig,
 )
```
Comment (gvanrossum) [inline python/semantic_kernel/kernel_extensions/inline_function_definitions.py:13]: Not for this PR, but this formatting is a hint that your module names are too long or your namespaces are too deeply nested. :-)

File: python/semantic_kernel/utils/validation.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    :raises ValueError: If the skill name is invalid.
+    """
+    if not value:
+        raise ValueError("The skill name cannot be `None` or empty")
+
+    SKILL_NAME_REGEX = r"^[0-9A-Za-z_]*$"
+    if not re_match(SKILL_NAME_REGEX, value):
+        raise ValueError(
+            f"Invalid skill name: {value}. Skill "
+            f"names must match the regex: {SKILL_NAME_REGEX}"
```
Comment (gvanrossum) [inline python/semantic_kernel/utils/validation.py:25]: This is a bit weak as a user-facing error; why not explain that skill names may only contain ASCII letters, digits, and underscores?

Comment (jjhenkel) [reply python/semantic_kernel/utils/validation.py]: Agreed, edited to match your suggestion.

File: python/semantic_kernel/kernel_extensions/inline_function_definitions.py (hunk 1 above)
Comment (jjhenkel) [reply python/semantic_kernel/kernel_extensions/inline_function_definitions.py:13]: Agreed, I think after we get through removing some more indirection and config classes, we could take a pass and clean up module names & nesting!

**Comment by mkarle** [comment]
Looks good to me!

**Review by mkarle — APPROVED** [review]

**Pull request merged**